dissect/target/containers/qcow2.py

```diff
@@ -37,4 +37,4 @@ def tell(self) -> int:
         return self.qcow2.tell()
 
     def close(self) -> None:
-        pass
+        self.qcow2.fh.close()
```

Comment:
I understand that from the library as a whole perspective and I mostly agree.
If I understand you correctly, I should open a PR in each of the underlying format implementations (e.g. qcow2 in `dissect.hypervisor`) and implement close over there right? 
BTW, I tried to trace the close call of ASDF and EWF, and noticed that they are deriving it from `dissect.utils.AlignedStream` which is also doing nothing :) 

https://github.com/fox-it/dissect.util/blob/fd72425d224b2220b3546d9f149103ae12479469/dissect/util/stream.py#L169-L171

Due to the fact every disk implementation is deriving from `dissect.utils.AlignedStream`, I suggest we create consistency between EWF, ASDF and the rest, and do the same to prepare for overriding close method in each disk implantation. 

```suggestion
        self.qcow2.close()
```

Reply:
I think that before we get into fixing the issue of closing the file handles, we need to decide where should we open the file handle. Now the inconsistency between those 3 containers, and other container implementation is just making more confusion in the code and whose the one managing the resources. 

IMO It should be in the container implementation due to how `SplitContainer` is implemented, but I might miss some other dissect use cases of disk implementation that such change might break.
 
And I totally agree that the file handle should be instance variable everywhere.

Reply:
When implementing it in the `Container` class, there are still the 3 execptions that need to do it in the underlying implementation (`AsdfContainer`, `VhdxContainer` and `VmdkContainer`). So for me it makes more sense to do it in the underlying implementation.

This would mean accepting `Path`s (and opening them) as well as open file handles. But that would be ok as we want the underlying implementations to be usable on their own. `SplitContainer` and `RawContainer` do not have an underlying implementation, so those two will need to implement the `close()` necessarily in their own class.

We could sub-class `AlignedStream` into something called `AlignedFileStream` which would implement the `close()` and an `__init__()` that deals with file `Path`(s) and open file handle(s) and keeps hem around.

One other thing that `close()` and `__init__()` need to do is to keep track of whether the file was opened by the instance or if an already open file handle was passed.

As `AlignedStream` (and also `Container`) is sub-classed from `io.RawIOBase` (resp. `io.IOBase`), `close()` will be called on destruction of an instance. Which is something that happens when the instance goes out of scope. If the instance was passed an already open file handle, it would be closed while the caller would not be aware of the closing, which would lead to all kinds of issues.